From ab1f09bde9fe4971d53ab0ea2fef25ecb0251feb Mon Sep 17 00:00:00 2001
From: Jaeyoon Jung <jaeyoon.jung@lge.com>
Date: Tue, 12 Jun 2018 20:25:29 +0900
Subject: [PATCH] Don't send synthetic hover event on a frame update

This reverts the commit 6f84a09 partially which generates a synthetic
hover event on a frame update. Such hover event has never been expected
in webOS and messes up the existing logic.

Change-Id: I7b855d27d1d44f2fab240707ef1ec79cdbcff6e5
Upstream-Status: Inappropriate [webOS specific]
---
 src/quick/items/qquickwindow.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/quick/items/qquickwindow.cpp b/src/quick/items/qquickwindow.cpp
index 6fd52a4c6..8be3911e7 100644
--- a/src/quick/items/qquickwindow.cpp
+++ b/src/quick/items/qquickwindow.cpp
@@ -2235,6 +2235,9 @@ void QQuickWindowPrivate::flushFrameSynchronousEvents()
             ut->startAnimations();
     }
 
+    // In webOS we already have the alternative to the issue that this
+    // wanted to address and thus skipping this part won't break anything.
+#if 0
     // Once per frame, if any items are dirty, send a synthetic hover,
     // in case items have changed position, visibility, etc.
     // For instance, during animation (including the case of a ListView
@@ -2246,6 +2249,7 @@ void QQuickWindowPrivate::flushFrameSynchronousEvents()
         if (!delivered)
             clearHover(); // take care of any exits
     }
+#endif
 }
 
 QQuickPointerEvent *QQuickWindowPrivate::queryPointerEventInstance(QQuickPointerDevice *device, QEvent::Type eventType) const
